@kiva/kv-components/vue/KvCheckbox.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+	<div>
+		<label
+			class="tw-inline-flex tw-items-center"
+			:class="{ 'tw-opacity-low': disabled }"
+			:for="generatedId"
+		>
+			<input
+				v-bind="$attrs"
+				:id="generatedId"
+				ref="checkboxRef"
+				class="tw-peer tw-appearance-none tw-w-max"
+				type="checkbox"
+				:checked="checked"
+				:value="value"
+				:disabled="disabled"
+				v-on="inputListeners"
+				@change.prevent="onChange"
+			>
+			<!-- checkbox square background -->
+			<div
+				class="
+					tw-w-3 tw-h-3 tw-mr-2
+					tw-flex-shrink-0
+					tw-rounded-sm
+					tw-border
+					tw-flex tw-justify-center tw-items-center tw-overflow-hidden
+					tw-transition-all tw-duration-100
+					peer-focus-visible:tw-ring-2 peer-focus-visible:tw-ring-action
+					tw-bg-white tw-border-gray-500
+					peer-checked:tw-bg-action peer-checked:tw-border-action
+				"
+			>
+				<!-- checkbox icon  -->
+				<svg
+					v-if="checked"
+					class="tw-w-1.5 tw-h-auto"
+					viewBox="0 0 12 9"
+					fill="none"
+					xmlns="http://www.w3.org/2000/svg"
+				>
+					<!-- eslint-disable max-len -->
+					<path
+						d="M3.99975 6.79988L1.66642 4.46655C1.40642 4.20655 0.993086 4.20655 0.733086 4.46655C0.473086 4.72655 0.473086 5.13988 0.733086 5.39988L3.52642 8.19322C3.78642 8.45322 4.20642 8.45322 4.46642 8.19322L11.5331 1.13322C11.7931 0.873216 11.7931 0.459883 11.5331 0.199883C11.2731 -0.0601172 10.8598 -0.0601172 10.5998 0.199883L3.99975 6.79988Z"
+						fill="white"
+					/>
+					<!-- eslint-enable max-len -->
+				</svg>
+			</div>
+			<!-- label -->
+			<div class="tw-flex-1 peer-focus-visible:tw-ring-2 peer-focus-visible:tw-ring-action">
+				<slot></slot>
+			</div>
+		</label>
+	</div>
+</template>
+
+<script>
+import { nanoid } from 'nanoid';
+
+/**
+ * Use as you would an <input type="checkbox" />
+ * Does not handle 'indeterminate' state at this time.
+*/
+
+export default {
```

Comment:
since you're applying $attrs and $listeners to the input component, I think you'll also want to set `inheritAttrs: false` in the component options so that the attributes and listeners don't get applied to both the root div and the input element https://vuejs.org/v2/api/#inheritAttrs

Reply:
thank you!